pkg/cli/admin/node/logs.go

```diff
@@ -112,28 +121,35 @@ func NewCmdLogs(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.
 		},
 	}
 
-	cmd.Flags().StringVar(&o.Path, "path", o.Path, "Retrieve the specified path within the node's /var/logs/ folder. The 'journal' value will allow querying the journal on supported operating systems.")
-
-	cmd.Flags().StringSliceVarP(&o.Units, "unit", "u", o.Units, "Return log entries from the specified unit(s). Only applies to node journal logs.")
-	cmd.Flags().StringVarP(&o.Grep, "grep", "g", o.Grep, "Filter log entries by the provided regex pattern. Only applies to node journal logs.")
+	cmd.Flags().StringVar(&o.Path, "path", o.Path, "Retrieve the specified path within the node's /var/logs/ folder. The 'journal' value will allow querying the services on supported operating systems.")
+	cmd.Flags().StringSliceVarP(&o.Query, "query", "q", o.Units, "Specifies services(s) or files from which to return logs")
+	cmd.Flags().StringSliceVarP(&o.Units, "unit", "u", o.Units, "Return log entries from the specified services(s) Only applies to node service logs.")
+	cmd.Flags().StringVarP(&o.Grep, "grep", "g", o.Grep, "Filter log entries by the provided regex pattern. Only applies to node service logs.")
 	cmd.Flags().BoolVar(&o.GrepCaseSensitive, "case-sensitive", o.GrepCaseSensitive, "Filters are case sensitive by default. Pass --case-sensitive=false to do a case insensitive filter.")
-	cmd.Flags().StringVar(&o.SinceTime, "since", o.SinceTime, "Return logs after a specific ISO timestamp or relative date. Only applies to node journal logs.")
-	cmd.Flags().StringVar(&o.UntilTime, "until", o.UntilTime, "Return logs before a specific ISO timestamp or relative date. Only applies to node journal logs.")
-	cmd.Flags().IntVar(&o.Boot, "boot", o.Boot, " Show messages from a specific boot. Use negative numbers, allowed [-100, 0], passing invalid boot offset will fail retrieving logs. Only applies to node journal logs.")
-	cmd.Flags().StringVarP(&o.Output, "output", "o", o.Output, "Display journal logs in an alternate format (short, cat, json, short-unix). Only applies to node journal logs.")
-	cmd.Flags().IntVar(&o.Tail, "tail", o.Tail, "Return up to this many lines (not more than 100k) from the end of the log. Only applies to node journal logs.")
-
+	cmd.Flags().StringVar(&o.Since, "since", o.Since, "Return logs after a specific ISO timestamp or relative date. Only applies to node service logs.")
+	cmd.Flags().StringVar(&o.SinceTime, "since-time", o.Since, "Return logs after a specific RFC3339 timestamp. Only applies to node service logs.")
+	cmd.Flags().StringVar(&o.Until, "until", o.Until, "Return logs before a specific ISO timestamp or relative date. Only applies to node service logs.")
+	cmd.Flags().StringVar(&o.UntilTime, "until-time", o.Since, "Return logs until a specific RFC3339 timestamp. Only applies to node service logs.")
```

Comment:
Drop `--since-time` and `--until-time` they are not needed, we should handle the translation in oc to use either of the two.

Reply:
Like I mentioned in Slack, converting `since` to `sinceTime` and `until` to `untilTime` is a bit tough. since and until can be both a relative time or ISO time stamp. `sinceTime` and `untilTime` only supports RFC3339 timestamps. I will drop `--since-time` and `--until-time` for now.